Detectors/AOD/src/AODProducerWorkflowSpec.cxx

```diff
@@ -350,6 +350,42 @@ void AODProducerWorkflowDPL::fillTrackTablesPerCollision(int collisionID,
   }
 }
 
+void AODProducerWorkflowDPL::fillIndexTablesPerCollision(const o2::dataformats::VtxTrackRef& trackRef, const gsl::span<const GIndex>& GIndices)
+{
+  //  for (int src = GIndex::NSources; src--;) {
+  for (int src : {GIndex::Source::MFTMCH, GIndex::Source::MCH, GIndex::Source::MFT}) {
+    int start = trackRef.getFirstEntryOfSource(src);
+    int end = start + trackRef.getEntriesOfSource(src);
+    for (int ti = start; ti < end; ti++) {
+      auto& trackIndex = GIndices[ti];
+      if (GIndex::includesSource(src, mInputSources)) {
+        if (src == GIndex::Source::MFT) {
+          if (trackIndex.isAmbiguous() && mGIDToTableMFTID.find(trackIndex) != mGIDToTableMFTID.end()) {
+            continue;
+          }
+
+          mGIDToTableMFTID.emplace(trackIndex, mIndexMFTID);
+          //          mIndexTableMFT.emplace(trackIndex.getIndex(), mIndexMFTID);
+          mIndexTableMFT[trackIndex.getIndex()] = mIndexMFTID;
+          mIndexMFTID++;
+
+        } else if (src == GIndex::Source::MCH || src == GIndex::Source::MFTMCH) {
+          if (trackIndex.isAmbiguous() && mGIDToTableFwdID.find(trackIndex) != mGIDToTableFwdID.end()) {
+            continue;
+          }
+
+          mGIDToTableFwdID.emplace(trackIndex, mIndexFwdID);
+          if (src == GIndex::Source::MCH) {
+            //            mIndexTableFwd.emplace(trackIndex.getIndex(), mIndexFwdID);
+            mIndexTableFwd[trackIndex.getIndex()] = mIndexFwdID;
+          }
+          mIndexFwdID++;
```

Comment:
why do you update mIndexTableFwd for MCH standalone only but always increment mIndexFwdID?

Reply:
I think here we only need the map to retrieve the correct indices for MCH standalone. But I increment mIndexFwdID so that we have the correct index in the fwdtrack table, i.e. we count all fwdtracks not only MCH standalone.